### PR TITLE
[generate_opcheck_tests] tests should ignore meta/FakeTensors

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109641
* #109640
* #109639
* #109638
* #109637

These tests generally don't work on meta tensors because they need to
compare the data of the Tensors. For example, SchemaCheckMode errors out
if any inputs are meta or Fake because it needs to check their storages
to see if any mutation occurred and those do not have storages.

Test Plan:
- new tests

### CODE REVIEW

File: torch/testing/_internal/optests/generate_tests.py [hunk 1]
```diff
@@ -21,10 +21,20 @@
 )
 
 
+def is_abstract(tensor: torch.Tensor) -> bool:
+    if tensor.is_meta:
+        return True
+    if isinstance(tensor, torch._subclasses.fake_tensor.FakeTensor):
```
Comment (bdhirsh) [inline torch/testing/_internal/optests/generate_tests.py:27]: Now that subclasses can be fakeified, you probably want `fake_tensor.is_fake`

**Review by bdhirsh — APPROVED** [review]
Curiosity question: how did this come up? Is it because you're directly using the example inputs from internal test suites, and some of the internal tests happen to have example inputs that are meta tensors?

**Comment by zou3519** [comment]
> Curiosity question: how did this come up? Is it because you're directly using the example inputs from internal test suites, and some of the internal tests happen to have example inputs that are meta tensors?

Yeah. People add tests that their meta formulas are correct by running meta tensors through operators

**Review by soulitzer — APPROVED** [review]

**Pull request closed without merging**